From 5d3f72ca9b9525dfaac55be1cb5bd7392114f87b Mon Sep 17 00:00:00 2001
From: Jarek Kobus <jaroslaw.kobus@qt.io>
Date: Tue, 28 Dec 2021 10:30:52 +0100
Subject: [PATCH 095/132] Fix pattern type matching

Amends 3c747aafa44ac2d0b314a002d2672227bf6513e5

Pick-to: 6.3 6.2 5.15
Change-Id: I2753b627c269cbb50009ef5e9c2ffa274e0ccf5c
Reviewed-by: David Faure <david.faure@kdab.com>
Reviewed-by: Alex Richardson <arichardson.kde@gmail.com>
(cherry picked from commit 9453038921f190eb8582e559df889cbc243f9e3c)
---
 src/corelib/mimetypes/qmimeglobpattern.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/corelib/mimetypes/qmimeglobpattern.cpp b/src/corelib/mimetypes/qmimeglobpattern.cpp
index 3ab5bd4cb4..fa8f4c545d 100644
--- a/src/corelib/mimetypes/qmimeglobpattern.cpp
+++ b/src/corelib/mimetypes/qmimeglobpattern.cpp
@@ -97,7 +97,7 @@ QMimeGlobPattern::PatternType QMimeGlobPattern::detectPatternType(const QString
     if (!patternLength)
         return OtherPattern;
 
-    const bool starCount = pattern.count(QLatin1Char('*')) == 1;
+    const int starCount = pattern.count(QLatin1Char('*'));
     const bool hasSquareBracket = pattern.indexOf(QLatin1Char('[')) != -1;
     const bool hasQuestionMark = pattern.indexOf(QLatin1Char('?')) != -1;
 
@@ -109,10 +109,10 @@ QMimeGlobPattern::PatternType QMimeGlobPattern::detectPatternType(const QString
             // Patterns like "README*" (well this is currently the only one like that...)
             if (pattern.at(patternLength - 1) == QLatin1Char('*'))
                 return PrefixPattern;
-        }
-        // Names without any wildcards like "README"
-        if (starCount == 0)
+        } else if (starCount == 0) {
+            // Names without any wildcards like "README"
             return LiteralPattern;
+        }
     }
 
     if (pattern == QLatin1String("[0-9][0-9][0-9].vdr"))
-- 
2.35.1

